website/src/components/SearchPage/useQueryAsState.ts

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect, type Dispatch, type SetStateAction } from 'react';
+
+import type { OrderDirection } from '../../types/lapis.ts';
+import { NULL_QUERY_VALUE } from '../../utils/search.ts';
+
+export interface QueryState {
+    page?: string;
+    orderBy?: string;
+    order?: OrderDirection;
+    [key: string]: string | string[] | undefined;
+}
+
+const MAX_URL_LENGTH = 2000; // Set a maximum URL length threshold
+
+function parseSearchToDict(search: string): QueryState {
+    const urlParams = new URLSearchParams(search);
+    const dict: QueryState = {};
+
+    for (const key of new Set(urlParams.keys())) {
+        const values = urlParams.getAll(key);
+        dict[key] = values.length === 1 ? values[0] : values;
+    }
+    return dict;
+}
+
+function dictToSearchParams(dict: QueryState) {
+    const params = new URLSearchParams();
+    for (const [key, value] of Object.entries(dict)) {
+        if (Array.isArray(value)) {
+            value.forEach((v) => {
+                params.append(key, v);
+            });
+        } else {
+            params.set(key, value ?? NULL_QUERY_VALUE);
```

Comment:
(again personally I prefer to be explicit about null vs undefined but nbd

Reply:
I'll just skip `undefined`s here. I think that makes more sense. We handle the "(blank)" case elsewhere anyway.

Reply:
`null` is not possible here. The type only allows `undefined`